core/trashcan.js

```diff
@@ -335,10 +402,80 @@ Blockly.Trashcan.prototype.close = function() {
  * Inspect the contents of the trash.
  */
 Blockly.Trashcan.prototype.click = function() {
-  var dx = this.workspace_.startScrollX - this.workspace_.scrollX;
-  var dy = this.workspace_.startScrollY - this.workspace_.scrollY;
-  if (Math.sqrt(dx * dx + dy * dy) > Blockly.DRAG_RADIUS) {
+  if (!this.hasBlocks) {
     return;
   }
-  console.log('TODO: Inspect trash.');
+
+  var xml = [];
+  for (var i = 0, text; text = this.contents[i]; i++) {
+    xml[i] = Blockly.Xml.textToDom(text).firstChild;
```

Comment:
Here's the current flow:
- A block is deleted
- The block is serialized to XML on the event
- The trashcan change listener grabs that XML
- The trashcan change listener modifies the XML (remove coordinates, etc)
- The trashcan change listener converts the XML to text and stores it as text
- The trashcan is clicked
- This code converts the text to XML
- The flyout goes from XML to blocks

You can just store the contents as XML instead of as text, to get rid of a couple of conversion steps.

Reply:
Ok I changed it over, and I'm getting some wierd behavior. Instead of saving the XML to the contents array, it seems like it's converting it into a block object? Which is cool but useless for comparing :/

![adding to contents behavior](https://user-images.githubusercontent.com/25440652/49839315-6d2c2780-fd63-11e8-8892-e134c071bc5c.jpg)

I did steal the converting to text thing from app inventor, so I assume they ran into the same problem and just decided to convert to text.

I left the logs in the latest commit so you can see if you so desire.

Reply:
Okay, let's go with text for simplicity.  Thanks!